internal/class.h

```diff
@@ -68,7 +70,9 @@ struct RClass {
     struct RBasic basic;
     VALUE super;
     struct rb_id_table *m_tbl;
-#if !USE_RVARGC
+#if USE_RVARGC
+    rb_alloc_func_t allocator;
```

Comment:
Can't we achieve the same result by moving the `allocator` field in `rb_classext_struct` to the first entry? That way we don't need these different preprocessor directives for whether `RVARGC` is enabled or not.